src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -206,53 +254,214 @@ public static FormulaValue Text(EvalVisitor runner, EvalVisitorContext context,
+                    return ResolveDateTimeFormatAmbiguities(format, dateTime, culture, cancellationToken);
             }
+        }
 
+        internal static string ExpandDateTimeFormatSpecifiers(string format, CultureInfo culture)
+        {
             var info = DateTimeFormatInfo.GetInstance(culture);
 
-            switch (format.ToLower().Trim('\''))
+            switch (format.ToLower())
             {
-                case "shortdatetime24":
+                case "'shortdatetime24'":
                     // TODO: This might be wrong for some cultures
                     return ReplaceWith24HourClock(info.ShortDatePattern + " " + info.ShortTimePattern);
-                case "shortdatetime":
+                case "'shortdatetime'":
                     // TODO: This might be wrong for some cultures
                     return info.ShortDatePattern + " " + info.ShortTimePattern;
-                case "shorttime24":
+                case "'shorttime24'":
                     return ReplaceWith24HourClock(info.ShortTimePattern);
-                case "shorttime":
+                case "'shorttime'":
                     return info.ShortTimePattern;
-                case "shortdate":
+                case "'shortdate'":
                     return info.ShortDatePattern;
-                case "longdatetime24":
+                case "'longdatetime24'":
                     return ReplaceWith24HourClock(info.FullDateTimePattern);
-                case "longdatetime":
+                case "'longdatetime'":
                     return info.FullDateTimePattern;
-                case "longtime24":
+                case "'longtime24'":
                     return ReplaceWith24HourClock(info.LongTimePattern);
-                case "longtime":
+                case "'longtime'":
                     return info.LongTimePattern;
-                case "longdate":
+                case "'longdate'":
                     return info.LongDatePattern;
-                case "utc":
+                case "'utc'":
                     return info.UniversalSortableDateTimePattern;
+                default:
+                    return format;
             }
+        }
+
+        private static string ReplaceWith24HourClock(string format)
+        {
+            format = Regex.Replace(format, "[hH]", "H");
+            format = Regex.Replace(format, "t+", "");
+
+            return format.Trim();
+        }
+
+        private static string ResolveDateTimeFormatAmbiguities(string format, DateTime dateTime, CultureInfo culture, CancellationToken cancellationToken)
+        {
+            format = ReplaceDoubleQuotedStrings(format, out var replaceList, cancellationToken);        
+            format = TokenizeDatetimeFormat(format, cancellationToken);
+            format = DetokenizeDatetimeFormat(format, dateTime, culture);
+            format = RestoreDoubleQuotedStrings(format, replaceList, cancellationToken);
 
             return format;
```

Comment:
>format [](http://example.com/codeflow?start=19&length=6)

[minor] What is returned isn't the format, but the actual formatted string; consider renaming it, or at least adding some comments to the method to clarify that.